src/Core/Commander/Providers/WMS_Provider.js

```diff
@@ -30,39 +32,235 @@ define('Core/Commander/Providers/WMS_Provider', [
          * @param {String} options.format: image format (default: format/jpeg)
          * @returns {Object@call;create.url.url|String}
          */
-        function WMS_Provider(options) {
+        function WMS_Provider(/*options*/) {
             //Constructor
 
             Provider.call(this, new IoDriver_XBIL());
             this.cache = CacheRessource();
             this.ioDriverImage = new IoDriver_Image();
             this.ioDriverXML = new IoDriverXML();
 
-            this.baseUrl = options.url || "";
-            this.layer = options.layer || "";
-            this.format = defaultValue(options.format, "image/jpeg");
-            this.srs = options.srs || "";
-            this.width = defaultValue(options.width, 256);
-            this.height = defaultValue(options.height, 256);
+            this._ready       = false;
+
+            this.getTextureFloat;
+
+            if(this.support)
+                this.getTextureFloat = function(){return new THREE.Texture();};
+            else
+                this.getTextureFloat = function(buffer){
+
+                    // Start float to RGBA uint8
+                    //var bufferUint = new Uint8Array(buffer.buffer);
+                    // var texture = new THREE.DataTexture(bufferUint, 256, 256);
+
+                    var texture = new THREE.DataTexture(buffer, 256, 256, THREE.AlphaFormat, THREE.FloatType);
+
+                    texture.needsUpdate = true;
+                    return texture;
+
+                };
         }
 
         WMS_Provider.prototype = Object.create(Provider.prototype);
 
         WMS_Provider.prototype.constructor = WMS_Provider;
 
+        WMS_Provider.prototype.url = function(bbox,layer) {
+            return this.customUrl(layer.customUrl, bbox);
+        };
+        WMS_Provider.prototype.customUrl = function(url,coord)
+        {
+
+            //convert radian to degree, lon is added a offset of Pi
+            //to align axis to card center
+            //var bbox = (coord.minCarto.longitude - Math.PI)* 180 / Math.PI + "," + coord.minCarto.latitude* 180 / Math.PI + "," +
+            //           (coord.maxCarto.longitude - Math.PI )*180/ Math.PI + "," + coord.maxCarto.latitude* 180 / Math.PI;
+
+            var bbox = coord.minCarto.latitude * 180.0 / Math.PI +
```

Comment:
As i said, origin of wms layers is at map center, because in WGS84 the prime meridian pass through GreenWich, which is not the case in our itowns by now. I have open an issue some months ago but it's not resolved yet. This problem touch to every module, so for now we can not move it to core, but one time core is modified, we can delete this code. #19